src/check/parse/Parser.zig

```diff
@@ -534,18 +537,10 @@ pub fn parseStmt(self: *Parser) ?IR.NodeStore.StatementIdx {
                 } });
                 return statement_idx;
             } else {
-                // If not a decl
-                const expr = self.parseExpr();
-                const statement_idx = self.store.addStatement(.{ .expr = .{
-                    .expr = expr,
-                    .region = .{ .start = start, .end = start },
-                } });
-                if (self.peek() == .Newline) {
-                    self.advance();
-                }
-                return statement_idx;
+                // continue to parse final expression
             }
```

Comment:
can we remove the empty else? Maybe put a comment before the if for clarity if it is needed?

Reply:
Alternatively, make parsing the final expression a function and `return parseFinalExpr(...)` in each else here.

Reply:
I thought including the else and a comment was more explicit that it was falling through to the last part. 

Happy to look at this in a follow up PR. I was removing some duplicate code here, but it's might be better to revert my change and leave it as it was.

Reply:
To be fair, might have just looked odd due to github pr review diff. I would lead to pulling it out into a function.

Ah yeah, looking at the raw source, this looks good. I think the diff view just made it look really spread out and unrelated.

That said, I think I would still prefer the final expression be a separate function. So it would be `else { return parseFinalExpr(...); }` same with the `else => {...}` branch of the switch. Then the switch will truly handle all cases with no fallthrough. Makes sure that someone doesn't accidentally fall through to the final expression parsing when they didn't mean to.